### PR TITLE
Cascading PID

### PR DESCRIPTION
**Pull request by matthew-reynolds**
# Pull Request

Adds general structure for internal PID controllers for cascading PID on voltage-controlled (simple) motor controllers.

 - Add syntax for specifying PID gains to yaml config file
 - Add PID gain switching on hardware interface change (ie different gains for pos, vel, & effort control)
 - Use PID controller to calculate output value, in volts, for the motors
 - Control whether to register command handles for specific hardware interfaces based on whether gains for that interface are specified.

The actual PID controller is untested and probably has bugs. However. the main intent of this PR is to get the structure of parsing gains from the file, switching between different gains as the hardware interface changes, and using the controller to calculate the output. This will set us up for adding smart speed controllers.

Debugging and fixing the real PID controllers will happen once we have a physical test fixture to debug with. Since we only use smart speed controllers in our robots, this is not critical and can therefore be pushed back and deprioritized slightly.

## Contribution Checklist

- [x] I have read the [contributing](https://github.com/uwreact/frc_control/blob/melodic-devel/CONTRIBUTING.md) guide.
- [x] I have referenced relevant issues.
- [x] I have labeled accordingly.
- [x] I have detailed my changes as much as possible.

## Change Checklist

- [ ] I have added tests.
- [x] I have added necessary documentation.
- [x] I have auto formatted using clang-format or yapf.

### CODE REVIEW

File: frc_robot_hw/src/frc_robot_hw_real.cpp [hunk 1]
```diff
@@ -490,6 +494,46 @@ bool FRCRobotHWReal::init(ros::NodeHandle& root_nh, ros::NodeHandle& robot_hw_nh
...
+
+  FRCRobotHW::doSwitch(start_list, stop_list);
+
+  using Mode = MultiPIDController::Mode;
+
+  // Disable PID for simple speed controllers claimed by stopping controllers
+  for (const auto& controller : stop_list) {
+    for (const auto& claimed : controller.claimed_resources) {
+      for (const auto& resource : claimed.resources) {
+        if (simple_speed_controller_pids_.find(resource) == simple_speed_controller_pids_.end())
```
Comment (ghost) [inline frc_robot_hw/src/frc_robot_hw_real.cpp:508]: why not
```
if (... != ...) {
  ...
}
```
instead of `continue`?

**Review by ghost — APPROVED** [review]

File: frc_robot_hw/src/frc_robot_hw_real.cpp (hunk 1 above)
Comment (matthew-reynolds) [reply frc_robot_hw/src/frc_robot_hw_real.cpp:508]: Removes a level of indentation in an already very indented block

Comment (ghost) [reply frc_robot_hw/src/frc_robot_hw_real.cpp:508]: alright :+1:

Comment (matthew-reynolds) [reply frc_robot_hw/src/frc_robot_hw_real.cpp:508]: In this case I suppose it's not a big deal, but in the starting controllers block it makes a difference, so I repeated it here for consistency

**Pull request merged**